src/ServiceManagement/Network/Commands.Network/Gateway/NewAzureVirtualNetworkGateway.cs

```diff
@@ -48,9 +54,47 @@ public class NewAzureVirtualNetworkGatewayCommand : NetworkCmdletBase
         [Parameter(Position = 7, Mandatory = false, HelpMessage = "Weight for routes learned from this BGP speaker")]
         public int PeerWeight { get; set; }
 
+        [Parameter(HelpMessage = "Do not confirm Azure Virtual Network Gateway Creation")]
+        public SwitchParameter Force { get; set; }
+
         public override void ExecuteCmdlet()
         {
-            WriteObject(Client.CreateVirtualNetworkGateway(VNetName, GatewayName, GatewayType, GatewaySKU, Location, VnetId, Asn, PeerWeight));
+            string gatewayType = "UltraPerformance";
+            if (GatewaySKU.Trim().Replace(" ", "").Equals(gatewayType, StringComparison.InvariantCultureIgnoreCase))
+            {
+                
+                ConfirmAction(
+               Force.IsPresent,
+               Resources.UltraPerformanceGatewayWarning,
+               Resources.UltraPerformanceGatewayContinue,
+               GatewaySKU + " " + GatewayType,
+               () =>
+                   {
+                       var gateway = Client.CreateVirtualNetworkGateway(
+                           VNetName,
+                           GatewayName,
+                           GatewayType,
+                           GatewaySKU,
+                           Location,
+                           VnetId,
+                           Asn,
+                           PeerWeight);
+                       WriteVerboseWithTimestamp(Resources.UltraPerformanceGatewaySucceed);
+                   WriteObject(gateway);
+               });
+            }
+            else
+            {
+                WriteObject(Client.CreateVirtualNetworkGateway(
```

Comment:
This also needs to be protected with ConfirmAction.  The easiest way would be to rewrite this to use a single invocation of ConfirmAction:

```
Force.IsPresent,
Resources.UltraPerformanceGatewayWarning,
 "Create Gateway",
GatewaySKU + " " + GatewayType,
 () =>
   {
     var gateway = Client.CreateVirtualNetworkGateway(
     VNetName,
     GatewayName,
     GatewayType,
     GatewaySKU,
     Location,
     VnetId,
     Asn,
     PeerWeight);
     WriteVerboseWithTimestamp(Resources.UltraPerformanceGatewaySucceed);
     WriteObject(gateway);
  },
  () => GatewaySKU.Trim().Replace(" ", "").Equals(gatewayType,StringComparison.InvariantCultureIgnoreCase));
```